modules/_labs/button/react/lib/deprecated_Button.tsx

```diff
@@ -0,0 +1,182 @@
+          height: '24px',
+          padding: `0 ${canvas.spacing.m}`,
+          minWidth: '80px',
+          maxWidth: '200px',
+          fontSize: '13px',
+          fontWeight: 500,
+        };
+      case ButtonSize.Small:
+        return {
+          height: '18px',
+          padding: `0 ${canvas.spacing.xxs}`,
+          minWidth: '56px',
+          maxWidth: '120px',
+          fontSize: '10px',
+          fontWeight: 500,
+        };
+    }
+  },
+  ({variant}) => {
+    let buttonColors;
+    switch (variant) {
+      case DeprecatedButtonVariant.Primary:
+        buttonColors = canvas.buttonColors.primary;
+        break;
+      case DeprecatedButtonVariant.Secondary:
+      default:
+        buttonColors = canvas.buttonColors.secondary;
+        break;
+      case DeprecatedButtonVariant.Delete:
+        buttonColors = {
+          ...canvas.buttonColors.delete,
+          focusBorder: canvas.colors.cinnamon500,
+          activeBorder: canvas.colors.cinnamon500,
+        };
+        break;
+    }
+
+    if (!buttonColors) {
+      return {};
+    }
+
+    const baseStyles = {
+      backgroundColor: buttonColors.background,
+      borderColor: buttonColors.border,
+      color: buttonColors.text,
+    };
+
+    const hoverStyles = {
+      ':hover': {
+        backgroundColor: buttonColors.hoverBackground,
+        borderColor: buttonColors.hoverBorder,
+        color: buttonColors.hoverText,
+      },
+    };
+
+    const activeStyles = {
+      ':active, :focus:active, :hover:active': {
+        backgroundColor: buttonColors.activeBackground,
+        borderColor: buttonColors.activeBorder,
+        color: buttonColors.activeText,
+      },
+    };
+
+    return {
+      ...baseStyles,
+      ':focus': {
+        backgroundColor: buttonColors.focusBackground,
+        borderColor: buttonColors.focusBorder,
+        color: buttonColors.focusText,
+      },
+
+      ...activeStyles,
+      ...hoverStyles,
+      ':disabled, :active:disabled, :focus:disabled, :hover:disabled': {
+        backgroundColor: buttonColors.disabledBackground,
+        borderColor: buttonColors.disabledBorder,
+        color: buttonColors.disabledText,
+      },
+      '&:not([disabled])': {
+        '&:focus': {
+          borderColor: buttonColors.focusBorder,
+          ...focusRing(2, 0),
+        },
+        '&:active': {
+          borderColor: buttonColors.activeBorder,
+          ...focusRing(2, 0),
+        },
+      },
+      ...mouseFocusBehavior({
+        '&:focus': {
+          ...baseStyles,
+          outline: 'none',
+          boxShadow: 'none',
+          animation: 'none',
+          ...hoverStyles,
+          ...activeStyles,
+        },
+      }),
+    };
+  }
+);
+
+const DeprecatedButton = (props: DeprecatedButtonProps) => {
+  const {variant, size, buttonRef, children, ...elemProps} = props;
+
+  React.useEffect(() => {
+    console.warn('This component is now deprecated, consider using the new Button component');
```

Comment:
since this is in labs and we have one outside of labs, should we make that distinction?

Reply:
This is only in labs to make the comparison in this PR clearer. I'm going to replace the old button with this one right after it's merged. Most people won't see this change because it's in the `prerelease/v4` branch.